persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/interceptors/ProxyProtocolQueryInterceptor.java

```diff
@@ -50,91 +120,158 @@ public void initialize() {}
+      case "rpc_address":
+        return InetAddressType.instance.decompose(publicAddress);
+      case "cluster_name":
+        return UTF8Type.instance.decompose(DatabaseDescriptor.getClusterName());
+      case "cql_version":
+        return UTF8Type.instance.decompose(QueryProcessor.CQL_VERSION.toString());
+      case "data_center":
+        return UTF8Type.instance.decompose(DatabaseDescriptor.getLocalDataCenter());
+      case "host_id":
+        return UUIDType.instance.decompose(UUID.nameUUIDFromBytes(publicAddress.getAddress()));
+      case "native_protocol_version":
+        return UTF8Type.instance.decompose(String.valueOf(ProtocolVersion.CURRENT.asInt()));
+      case "partitioner":
+        return UTF8Type.instance.decompose(
+            DatabaseDescriptor.getPartitioner().getClass().getName());
+      case "rack":
+        return UTF8Type.instance.decompose(DatabaseDescriptor.getLocalRack());
+      case "release_version":
+        return UTF8Type.instance.decompose(ProductVersion.getReleaseVersion().toString());
+      case "schema_version":
+        return UUIDType.instance.decompose(StargateSystemKeyspace.SCHEMA_VERSION);
+      case "tokens":
+        return SetType.getInstance(UTF8Type.instance, false)
+            .decompose(
+                Collections.singleton(
+                    DatabaseDescriptor.getPartitioner().getMinimumToken().toString()));
+      case "native_transport_port": // Fallthrough intentional
+      case "native_transport_port_ssl":
+        return Int32Type.instance.decompose(publicPort);
+      case "storage_port":
+        return Int32Type.instance.decompose(DatabaseDescriptor.getStoragePort());
+      case "storage_port_ssl":
+        return Int32Type.instance.decompose(DatabaseDescriptor.getSSLStoragePort());
+      case "jmx_port":
+        return DatabaseDescriptor.getJMXPort()
+            .map(p -> Int32Type.instance.decompose(p))
+            .orElse(null);
+      default:
+        return null;
+    }
+  }
+
+  /**
+   * Builds a row using the {@link CQLStatement}'s result metadata. This doesn't handles special
+   * cases like aggregates, null is returned in those cases, but it should be good enough for
+   * handling system tables.
+   *
+   * @param metadata
+   * @param publicAddress
+   * @param publicPort
+   * @return A list of {@link ByteBuffer} values for a system local/peers row.
+   */
+  private static List<ByteBuffer> buildRow(
+      ResultMetadata metadata, InetAddress publicAddress, int publicPort) {
+    List<ByteBuffer> row = Lists.newArrayListWithCapacity(metadata.names.size());
+    metadata.names.forEach(
+        column -> row.add(buildColumnValue(column.name.toString(), publicAddress, publicPort)));
+    return row;
+  }
+
+  private static class DefaultResolver implements Resolver {
+    @Override
+    public Set<InetAddress> resolve(String name) throws UnknownHostException {
+      return Arrays.stream(InetAddress.getAllByName(name)).collect(Collectors.toSet());
```

Comment:
You mentioned in the PR description:
> It also monitors the current state of the A records and if they change it generates the appropriate topology events for the CQL protocol

Where does that happen?

Reply:
Here: https://github.com/stargate/stargate/pull/267/files#diff-91bef8000571404fe1179e3c6e3e6cd6aed7fa2f9ba59ad198cd44e136dce632R168-R180

Reply:
Ok, I don't know how I could miss it, thanks.